pkg/streamingpromql/engine_test.go

```diff
@@ -1836,11 +1836,82 @@ func (t *timeoutTestingQueryTracker) Close() error {
 	return nil
 }
 
-func TestAnnotations(t *testing.T) {
+type annotationTestCase struct {
+	data                               string
+	expr                               string
+	expectedWarningAnnotations         []string
+	expectedInfoAnnotations            []string
+	skipComparisonWithPrometheusReason string
+}
+
+func runAnnotationTests(t *testing.T, testCases map[string]annotationTestCase) {
 	startT := timestamp.Time(0).Add(time.Minute)
 	step := time.Minute
 	endT := startT.Add(2 * step)
 
+	opts := NewTestEngineOpts()
+	mimirEngine, err := NewEngine(opts, NewStaticQueryLimitsProvider(0), stats.NewQueryMetrics(nil), log.NewNopLogger())
+	require.NoError(t, err)
+	prometheusEngine := promql.NewEngine(opts.CommonOpts)
+
+	const prometheusEngineName = "Prometheus' engine"
+	engines := map[string]promql.QueryEngine{
+		"Mimir's engine": mimirEngine,
+
+		// Compare against Prometheus' engine to verify our test cases are valid.
+		prometheusEngineName: prometheusEngine,
+	}
+
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			store := promqltest.LoadedStorage(t, "load 1m\n"+strings.TrimSpace(testCase.data))
+			t.Cleanup(func() { _ = store.Close() })
+
+			queryTypes := map[string]func(engine promql.QueryEngine) (promql.Query, error){
+				"range": func(engine promql.QueryEngine) (promql.Query, error) {
+					return engine.NewRangeQuery(context.Background(), store, nil, testCase.expr, startT, endT, step)
+				},
+				"instant": func(engine promql.QueryEngine) (promql.Query, error) {
+					return engine.NewInstantQuery(context.Background(), store, nil, testCase.expr, startT)
+				},
+			}
+
+			for queryType, generator := range queryTypes {
+				t.Run(queryType, func(t *testing.T) {
+					results := make([]*promql.Result, 0, 2)
+
+					for engineName, engine := range engines {
+						if engineName == prometheusEngineName && testCase.skipComparisonWithPrometheusReason != "" {
+							t.Logf("Skipping comparison with Prometheus' engine: %v", testCase.skipComparisonWithPrometheusReason)
+							continue
+						}
+
+						query, err := generator(engine)
+						require.NoError(t, err)
+						t.Cleanup(query.Close)
+
+						res := query.Exec(context.Background())
+						require.NoError(t, res.Err)
+						results = append(results, res)
+
+						_, infos := res.Warnings.AsStrings(testCase.expr, 0, 0)
+						//require.ElementsMatch(t, testCase.expectedWarningAnnotations, warnings)
```

Comment:
Should this be commented out?

Reply:
You're finding all my wip I accidentally committed, thanks! (I had like 15 nonsensical commits that I squashed :-p)